src/vmshepherd_aws_drivers/aws_iaas_driver.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import aiobotocore
+from typing import Any, Dict, List
+from vmshepherd.iaas import AbstractIaasDriver, Vm, VmState
+
+
+class AwsIaaSDriver(AbstractIaasDriver):
+
+    async def list_vms(self, preset_name):
+        '''
+        List VMs by preset name
+        :arg present_name: string
+        '''
+        loop = asyncio.get_event_loop()
+        session = aiobotocore.get_session(loop=loop)
+        async with session.create_client('ec2') as client:
+            res = await client.describe_instances(
+                Filters=[
+                    {
+                        'Name': 'tag:aws:autoscaling:groupName',
```

Comment:
This method is only for filter instances belonging to autoscaling groups, we use here ec2 api (instead of autoscaling api) to retrieve ips of vms. In my opinion the only proper grouping is asg name here, and it takes place in aws_preset_driver. imho in this case simple the best.

Reply:
or in everyware (preset, iaas drivers) can be used aws ec2 api and then grouping can be realized by anything, but i'd like to know use case for that